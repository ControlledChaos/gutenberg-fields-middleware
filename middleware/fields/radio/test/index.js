import React from 'react';
import { render } from 'enzyme';
const { RadioControl } = wp.components;

describe( 'RadioControl', () => {
	const onChange = jest.fn();
	const wrapper = render(
		<RadioControl
			label="User type"
			help="The type of the current user"
			selected={ 'a' }
			options={ [
				{
					label: 'Author',
					value: 'a',
				},
				{
					label: 'Editor',
					value: 'e',
				},
			] }
			onChange={ onChange } /> );

	beforeEach( () => {
		onChange.mockClear();
	} );

	test( 'should render a radio Control', () => {
		expect( wrapper ).toMatchSnapshot();
	} );
} );
