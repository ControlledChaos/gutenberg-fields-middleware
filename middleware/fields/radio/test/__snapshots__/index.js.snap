// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioControl should render a radio Control 1`] = `
<div
  class="components-base-control components-radio-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="inspector-radio-control-0"
    >
      User type
    </label>
    <div
      class="components-radio-control__option"
    >
      <input
        aria-describedby="inspector-radio-control-0__help"
        checked=""
        class="components-radio-control__input"
        id="inspector-radio-control-0-0"
        name="inspector-radio-control-0"
        type="radio"
        value="a"
      />
      <label
        for="inspector-radio-control-0-0"
      >
        Author
      </label>
    </div>
    <div
      class="components-radio-control__option"
    >
      <input
        aria-describedby="inspector-radio-control-0__help"
        class="components-radio-control__input"
        id="inspector-radio-control-0-1"
        name="inspector-radio-control-0"
        type="radio"
        value="e"
      />
      <label
        for="inspector-radio-control-0-1"
      >
        Editor
      </label>
    </div>
  </div>
  <p
    class="components-base-control__help"
    id="inspector-radio-control-0__help"
  >
    The type of the current user
  </p>
</div>
`;
